app/signup.tsx

```diff
@@ -187,12 +198,15 @@ export default function SignupScreen() {
           <TouchableOpacity
             style={[
               styles.signUpButton,
-              !isFormValid && styles.signUpButtonDisabled,
+              (!isFormValid || signupMutation.isPending) &&
```

Comment:
this condition is used here and also bellow, create a constant above and use it in both places, so it is easier to read and also to update the logic to disable the button when needed